provider/infoblox/infoblox.go

```diff
@@ -140,48 +153,44 @@ func NewInfobloxProvider(infobloxConfig InfobloxConfig) (*InfobloxProvider, erro
 // Records gets the current records.
 func (p *InfobloxProvider) Records(ctx context.Context) (endpoints []*endpoint.Endpoint, err error) {
 	zones, err := p.zones()
+
 	if err != nil {
 		return nil, fmt.Errorf("could not fetch zones: %s", err)
 	}
 
 	for _, zone := range zones {
-		logrus.Debugf("fetch records from zone '%s'", zone.Fqdn)
+		endpointsTypeA := make([]*endpoint.Endpoint, 0)
 		var resA []ibclient.RecordA
+		var recordsNameFilter string
+		if len(p.recordsNameFilter) > 0 {
+			logrus.Debugf("fetch records with recordsNameFilter:'%s'", p.recordsNameFilter)
+			recordsNameFilter = p.recordsNameFilter
+		} else {
+			recordsNameFilter = ""
+		}
```

Comment:
What i was wondering.
Why are you creating a new variable?
Is the default for `p.recordsNameFilter` already `""`
And can you use this inside the new records?
```suggestion
		if len(p.recordsNameFilter) > 0 {
			logrus.Debugf("fetch records with recordsNameFilter:'%s'", p.recordsNameFilter)
		}
```